### PR TITLE
feat: web3 react integration

### PR DESCRIPTION
**Pull request by aliangm**
[ ] Before opening a pull request, please read the [contributing guidelines](https://github.com/pancakeswap/pancake-frontend/blob/master/CONTRIBUTING.md) first
[ ] If your PR is work in progress, open it as `draft`
[ ] Before requesting a review, all the checks need to pass
[ ] Explain what your PR does

### CODE REVIEW

**Review by hachiojidev — CHANGES_REQUESTED** [review]
I know this is a draft but your commits are not following Conventional Commits. Please refer to https://github.com/pancakeswap/pancake-frontend/blob/develop/CONTRIBUTING.md

**Comment by Chef-Chungus** [comment]
> I know this is a draft but your commits are not following Conventional Commits. Please refer to https://github.com/pancakeswap/pancake-frontend/blob/develop/CONTRIBUTING.md

https://docs.pancakeswap.finance/code/contributing#committing

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "private": true,
   "scripts": {
     "analyze": "source-map-explorer 'build/static/js/*.js'",
-    "start": "react-scripts start",
+    "start": "HOST=0.0.0.0 react-scripts start",
```
Comment (hachiojidev) [inline package.json:7]: Why was this changed?

File: package.json [hunk 2]
```diff
@@ -33,18 +33,27 @@
     "@binance-chain/bsc-use-wallet": "^0.8.1",
```
Comment (hachiojidev) [inline package.json:33]: Need to remove this.

File: src/.env.development [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+REACT_APP_GTAG = "GTM-PXLD3XW"
+
+# 10+ nodes balanced, US/EU
+REACT_APP_NODE_1 = "https://bsc-dataseed1.ninicoin.io"
+
+# 10+ nodes balanced, US/EU
+REACT_APP_NODE_2 = "https://bsc-dataseed1.defibit.io"
+
+# 10+ nodes balanced in each region, global
+REACT_APP_NODE_3 = "https://bsc-dataseed.binance.org"
```
Comment (hachiojidev) [inline src/.env.development:11]: Is your branch synced with `develop`? This file hasn't changed.

File: src/components/Web3ReactManager/index.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  justify-content: center;
+  height: 20rem;
+`
+
+const Message = styled.h2`
+  color: ${({ theme }) => theme.colors.primaryDark};
+`
+
+export default function Web3ReactManager({ children }: { children: JSX.Element }) {
+  const t = useI18n()
```
Comment (hachiojidev) [inline src/components/Web3ReactManager/index.tsx:25]: Please use `TranslateString`.

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (aliangm) [reply package.json]: It's not needed. I will fix it

File: package.json [hunk 5]
```diff
@@ -30,21 +30,30 @@
...
+    "@web3-react/walletconnect-connector": "^6.1.6",
+    "@web3-react/walletlink-connector": "^6.1.6",
     "bignumber.js": "^9.0.0",
     "chart.js": "^2.9.4",
+    "ethers": "^5.0.26",
     "lodash": "^4.17.20",
     "react": "^17.0.1",
     "react-chartjs-2": "^2.11.1",
     "react-countup": "^4.3.3",
+    "react-device-detect": "^1.15.0",
```
Comment (RabbitDoge) [inline package.json:56]: I don't think we need this

File: src/App.tsx [hunk 6]
```diff
@@ -41,35 +43,37 @@ const App: React.FC = () => {
...
-            <Route path="/staking">
-              <Redirect to="/pools" />
-            </Route>
-            <Route path="/syrup">
-              <Redirect to="/pools" />
-            </Route>
-            {/* 404 */}
-            <Route component={NotFound} />
-          </Switch>
+          <Web3ReactManager>
```
Comment (RabbitDoge) [inline src/App.tsx:46]: We have a provider component called in the index.tsx

**Comment by RabbitDoge** [comment]
@aliangm it seems that some changes are not related to your pr. I don't know how you manage to update your branch, but I strongly suggest your use rebase instead of merge, and try to do it every day to avoid conflicts.

Also, it seems that you blindly copied some code from the exchange. The exchange implementation is probably not the best one to use as a reference. It's better that you do your own implementation from scratch, based on the web3 react doc and we will reuse it on the exchange later, rather that copying the exchange.

**Comment by aliangm** [comment]
Yeah, while checking the codebase again, I also noticed some unnecessary code added
As I didn't have deep understanding for FE repo, I thought the code should be necessary
And I do rebase instead of merge every time with develop branch. I think the merge conflict errors appeared because I forgot to pull develop branch from upstream
I will keep eyes open on it

File: src/views/Nft/components/TransferNftModal.tsx [hunk 7]
```diff
@@ -51,21 +51,18 @@ const TransferNftModal: React.FC<TransferNftModalProps> = ({ nft, tokenIds, onSu
...
-          .on('error', () => {
-            console.error(error)
-            setError('Unable to transfer NFT')
-            setIsLoading(false)
-          })
+        setIsLoading(true)
+        const tx = pancakeRabbitsContract
+          .transferFrom(account, value, tokenIds[0], { from: account })
+        try {
+          tx.wait()
```
Comment (hachiojidev) [inline src/views/Nft/components/TransferNftModal.tsx:58]: Is there a reason to use this over async?

**Review by hachiojidev — CHANGES_REQUESTED** [review]
Nice work. There are a few minor issues and one big issue I noticed with BSC Wallet.

File: src/components/Web3ReactManager/index.tsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  // if the account context isn't active, and there's an error on the network context, it's an irrecoverable error
+  if (networkError) {
+    return (
+      <MessageWrapper>
+        <Message>{TranslateString(999, 'unknownError')}</Message>
+      </MessageWrapper>
+    )
+  }
```
[the remark this answers is not in the document]
Comment (aliangm) [reply src/components/Web3ReactManager/index.tsx:65]: Ok

File: src/components/Web3ReactManager/index.tsx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+import { useWeb3React } from '@web3-react/core'
+import styled from 'styled-components'
+import useI18n from 'hooks/useI18n'
+
+import PageLoader from 'components/PageLoader'
+import { network } from 'connectors'
+import useEagerConnect from 'hooks/useEagerConnect'
+import useInactiveListener from 'hooks/useInactiveListener'
+
+// export const NetworkContextName = 'NETWORK'
```
[the remark this answers is not in the document]
Comment (RabbitDoge) [reply src/components/Web3ReactManager/index.tsx:11]: @aliangm remove commented code plz

[the remark this answers is not in the document]
Comment (aliangm) [reply src/components/Web3ReactManager/index.tsx:11]: Ok

File: src/state/profile/getProfile.ts [hunk 10]
```diff
@@ -1,55 +1,49 @@
...
-      username,
-      nftAddress,
-      isActive,
-      nft,
-      team,
-    } as Profile
-  } catch (error) {
+  const profileContract = getContract(getPancakeProfileAddress(), pancakeProfileAbi)
+  const rabbitContract = getContract(getPancakeRabbitsAddress(), pancakeRabbitsAbi)
+  const profileApi = process.env.REACT_APP_API_PROFILE
```
[the remark this answers is not in the document]
Comment (aliangm) [reply src/state/profile/getProfile.ts:14]: The issue when we move those to out of scope is they're initiated with wrong provider.
As long as we can't use hooks, there's no way to detect provider changes

File: src/views/Nft/components/TransferNftModal.tsx (hunk 7 above)
Comment (aliangm) [reply src/views/Nft/components/TransferNftModal.tsx:58]: Line 58 seems wrong. It should be like `await tx.wait()`

File: package.json (hunk 5 above)
Comment (hachiojidev) [reply package.json:56]: This needs to be removed.

File: src/views/Profile/hooks/useGetProfileCosts.ts [hunk 11]
```diff
@@ -15,17 +16,26 @@ const useGetProfileCosts = () => {
...
+  
+        const res = await multicall(profileABI, calls)
 
         setCosts({
-          numberCakeToReactivate: new BigNumber(numberCakeToReactivate as string),
-          numberCakeToRegister: new BigNumber(numberCakeToRegister as string),
-          numberCakeToUpdate: new BigNumber(numberCakeToUpdate as string),
+          numberCakeToReactivate: new BigNumber(res[0].toString()),
+          numberCakeToRegister: new BigNumber(res[1].toString()),
+          numberCakeToUpdate: new BigNumber(res[2].toString()),
```
Comment (hachiojidev) [inline src/views/Profile/hooks/useGetProfileCosts.ts:38]: Please destructure this above on line 33.

File: src/state/profile/getProfile.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (hachiojidev) [reply src/state/profile/getProfile.ts:18]: I am referring to `process.env.REACT_APP_API_PROFILE`. It is written on build and shouldn't ever change.

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: src/views/Profile/hooks/useGetProfileCosts.ts (hunk 11 above)
Comment (aliangm) [reply src/views/Profile/hooks/useGetProfileCosts.ts:38]: Ok

**Review by bread-cn — APPROVED** [review]

File: src/react-app-env.d.ts [hunk 12]
```diff
@@ -1 +1,41 @@
 /// <reference types="react-scripts" />
+
+import { Web3Provider } from '@ethersproject/providers'
+
+declare module 'jazzicon' {
+  export default function (diameter: number, seed: number): HTMLElement
+}
```
Comment (hachiojidev) [inline src/react-app-env.d.ts:7]: What is this for?

File: src/views/Profile/components/AchievementRow/index.tsx [hunk 13]
```diff
@@ -42,25 +42,21 @@ const AchievementRow: React.FC<AchievementRowProps> = ({ achievement, onCollectS
...
-        setIsCollecting(false)
-        onCollectSuccess(achievement)
-        toastSuccess('Points Collected!')
-      })
-      .on('error', (error) => {
-        toastError('Error', error?.message)
-        setIsCollecting(false)
-      })
+  const handleCollectPoints = async () => {
+    const tx = pointCenterContract.getPoints(achievement.address, { from: account })
```
Comment (hachiojidev) [inline src/views/Profile/components/AchievementRow/index.tsx:49]: For calls like this, is it required to send `{ from: account }` ?

File: src/react-app-env.d.ts (hunk 12 above)
Comment (aliangm) [reply src/react-app-env.d.ts:7]: Hmm.. I'm not sure. Is it sth automatically generated code?

File: src/views/Profile/components/AchievementRow/index.tsx (hunk 13 above)
Comment (aliangm) [reply src/views/Profile/components/AchievementRow/index.tsx:49]: Yeah, the original codebase with web3.js was look like this
pointCenterContract.methods
      .getPoints(achievement.address)
      .send({ from: account })

It's required to send ({from: account})

File: src/state/achievements/helpers.ts [hunk 14]
```diff
@@ -24,22 +25,22 @@ export const getUserPointIncreaseEvents = async (account: string) => {
...
       {
-        id: event.args[2],
+        id: event.args[2].toString(),
         type: campaignMeta.type,
         address: event.address,
         title: getAchievementTitle(campaignMeta),
         description: getAchievementDescription(campaignMeta),
         badge: campaignMeta.badge,
-        points: Number(event.args[1]),
+        points: Number(event.args[1].toString()),
```
Comment (hachiojidev) [inline src/state/achievements/helpers.ts]: For the future please always deconstruct the variable. Using the index directly makes it virtually impossible without inspecting the return for other devs to understand what the variable is.

```
const [, points, campaignId] = event.args;
```

**Comment by RabbitDoge** [comment]
fix #318

File: src/state/achievements/helpers.ts (hunk 14 above)
Comment (aliangm) [reply src/state/achievements/helpers.ts]: Ok

File: src/views/Profile/ProfileCreation/ProfilePicture.tsx [hunk 15]
```diff
@@ -28,26 +28,22 @@ const ProfilePicture: React.FC = () => {
...
+  const handleApprove = async () => {
+    setIsApproving(true)
+    try {
+      const tx = await pancakeRabbitsContract.approve(getPancakeProfileAddress(), tokenId, { from: account })
+      await tx.wait()
+      setIsApproving(false)
+      setIsApproved(true)
+    } catch (error) {
+      toastError('Error', error?.message)
+      setIsApproving(false)
```
Comment (hachiojidev) [inline src/views/Profile/ProfileCreation/ProfilePicture.tsx:45]: Lets put `setIsApproving(false)` in a `finally` block.

File: src/views/Profile/ProfileCreation/UserName.tsx [hunk 16]
```diff
@@ -108,11 +106,12 @@ const UserName: React.FC = () => {
...
     try {
       setIsLoading(true)
-
       const provider = ethereum as any
+      const signer = provider.getSigner()
+
       const signature = provider?.bnbSign
         ? (await provider.bnbSign(account, userName))?.signature
-        : await web3.eth.personal.sign(userName, account, null) // Last param is the password, and is null to request a signature in the wallet
+        : await signer.signMessage(userName) // Last param is the password, and is null to request a signature in the wallet
```
Comment (hachiojidev) [inline src/views/Profile/ProfileCreation/UserName.tsx:114]: Doesn't seem this comment is relevant anymore. Please remove if that is the case.

**Review by RabbitDoge — APPROVED** [review]
Still a lot of improvements to be done, but let's go with this first iteration.

**Review by hachiojidev — APPROVED** [review]

File: src/hooks/useGetWalletNfts.ts [hunk 17]
```diff
@@ -51,24 +53,35 @@ const reducer = (state: State, action: Action) => {
...
+                  params: [tokenId.toString()],
+                },
+                {
+                  address: getPancakeRabbitsAddress(),
+                  name: 'tokenURI',
+                  params: [tokenId.toString()],
+                },
+              ]
+              const res = await multicall(pancakeRabbitsAbi, calls)
+              return [Number(res[0]), Number(tokenId.toString()), res[1]]
```
[the remark this answers is not in the document]
Comment (aliangm) [reply src/hooks/useGetWalletNfts.ts]: Fixed
